src/system/UsersModule/Twig/Extension/ProfileExtension.php

```diff
@@ -0,0 +1,154 @@
+     *   {{ username|profileLinkByUserName(class='classname') }}
+     *   Using profile.gif instead of username, no class
+     *   {{ username|profileLinkByUserName('image'='images/profile.gif') }}
+     *
+     * @param string  $userName  The users name
+     * @param string  $class     The class name for the link (optional)
+     * @param string  $image     Path to the image to show instead of the username (optional)
+     * @param integer $maxLength If set then user names are truncated to x chars
+     * @return string The output
+     */
+    public function profileLinkByUserName($userName, $class = '', $image = '', $maxLength = 0)
+    {
+        if (empty($userName)) {
+            return $userName;
+        }
+
+        return $this->determineProfileLink(null, $userName, $class, $image, $maxLength);
+    }
+
+    /**
+     * Internal function used by profileLinkByUserId() and profileLinkByUserName().
+     *
+     * @param integer $userId    The users uid
+     * @param string  $userName  The users name
+     * @param string  $class     The class name for the link (optional)
+     * @param string  $imagePath Path to the image to show instead of the username (optional)
+     * @param integer $maxLength If set then user names are truncated to x chars
+     * @return string The output
+     */
+    private function determineProfileLink($userId = null, $userName = null, $class = '', $imagePath = '', $maxLength = 0)
+    {
+        if (!isset($userId) && !isset($userName)) {
+            throw new \InvalidArgumentException();
+        }
+        if ($userId) {
+            $user = $this->userRepository->find($userId);
+        } else {
+            $user = $this->userRepository->findOneBy(['uname' => $userName]);
+        }
+        if (!$user) {
+            return $userId . $userName; // one or the other is empty
+        }
+
+        $userDisplayName = $this->profileModuleCollector->getSelected()->getDisplayName($user->getUid());
+        $class = !empty($class) ? ' class="' . htmlspecialchars($class, ENT_QUOTES) . '"' : '';
+
+        if (!empty($imagePath)) {
+            $show = '<img src="' . htmlspecialchars($imagePath, ENT_QUOTES) . '" alt="' . htmlspecialchars($userDisplayName, ENT_QUOTES) . '" />';
+        } elseif ($maxLength > 0) {
+            // truncate the user name to $maxLength chars
+            $length = strlen($userDisplayName);
+            $truncEnd = ($maxLength > $length) ? $length : $maxLength;
+            $show  = htmlspecialchars(substr($userDisplayName, 0, $truncEnd), ENT_QUOTES);
+        } else {
+            $show = htmlspecialchars($userDisplayName, ENT_QUOTES);
+        }
+        $href = $this->profileModuleCollector->getSelected()->getProfileUrl($user->getUid());
+
+        return '<a' . $class . ' title="' . ($this->translator->__('Profile')) . ': ' . htmlspecialchars($userDisplayName, ENT_QUOTES) . '" href="' . $href . '">' . $show . '</a>';
```

Comment:
Please return just the user display name if `$href` equals `#`. This keeps BC and avoids non-working links.